### ISSUE TITLE
docker network inspect missing info if no containers on that network are present on the manager

### ISSUE THREAD

**Issue by mcodd** [moby/moby#36620]
This is a bug filed as an update to #24186 

**Description**

On Docker 17.12.1 and previous versions (also tested 17.12 and 17.09), I am trying to use `docker network inspect -v` to gather a list of IP addresses of containers on various services that are attached to that network.  When I run the command against a master node of the swarm which has no containers running locally on it that are attached to the network I'm inspecting, I get output that does not include information about any Services, and other information about the network like the Created field have incorrect/missing data.

**Steps to reproduce the issue:**
1. Start up a stack with networks configured such that the only services attached to them run exclusively on workers, NOT managers.
2. Query that network with `docker network inspect -v`

**Describe the results you received:**

When querying such a network against a manager node:
```
$ docker --tlsverify --host tcp://omni-sandbox-manager-g1:2376 network inspect sandbox-swarm_dmz -v
[
    {
        "Name": "sandbox-swarm_dmz",
        "Id": "7fh2betl7y4qxc36xabdtnf4l",
        "Created": "2018-03-17T01:29:24.664686271Z",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "",
            "Options": null,
            "Config": null
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": null,
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4102",
            "encrypted": ""
        },
        "Labels": {
            "com.docker.stack.namespace": "sandbox-swarm"
        }
    }
]
$ 
```

**Describe the results you expected:**

When querying a worker node that is running a container on a service attached to that network:

```
$ docker --tlsverify --host tcp://omni-sandbox-proxyworker-g1:2376 network inspect sandbox-swarm_dmz -v
[
    {
        "Name": "sandbox-swarm_dmz",
        "Id": "7fh2betl7y4qxc36xabdtnf4l",
        "Created": "2018-03-16T21:29:25.180049372-04:00",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.5.0/24",
                    "Gateway": "10.0.5.1"
                }   
            ]   
        },  
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },  
        "ConfigOnly": false,
        "Containers": {
            "0abfba473b398d0afd01a54858c96a42272a7db70fc669229cb3e574dde3c3db": {
                "Name": "sandbox-swarm_metricproxy.3.81lu8uzik0hotx871zd99l0tk",
                "EndpointID": "141c883dcc93e1e3766e5d9505dff70e40c98fd8c01919600a05c965847c7320",
                "MacAddress": "02:42:0a:00:05:0e",
                "IPv4Address": "10.0.5.14/24",
                "IPv6Address": ""
            },  
[[[ TRUNCATING LONG OUTPUT ]]]
...
            "sandbox-swarm_xray-daemon": {
                "VIP": "10.0.5.22",
                "Ports": [],
                "LocalLBIndex": 269,
                "Tasks": [
                    {   
                        "Name": "sandbox-swarm_xray-daemon.3.kfue6oczjjp421s1fqxg6j0fq",
                        "EndpointID": "5cdf473c341b1b1e7898830aa5268a21c92e6e59a9ea9af72a987944c500a18d",
                        "EndpointIP": "10.0.5.25",
                        "Info": {
                            "Host IP": "172.24.35.206"
                        }
                    },
                    {   
                        "Name": "sandbox-swarm_xray-daemon.1.bauyq21f6y2q6nynjtxpyekqc",
                        "EndpointID": "77744d6d2fe9fd53481b342527d95a066c86df35fe7078ce7471cf447dea9158",
                        "EndpointIP": "10.0.5.23",
                        "Info": {
                            "Host IP": "172.24.35.206"
                        }
                    },
                    {   
                        "Name": "sandbox-swarm_xray-daemon.2.h63sfl06kswcrrm7nujogcvmm",
                        "EndpointID": "a4a61dfa75adbf73505b4b44f93cab826bfe60193682b7a7d74960ebcaf461b1",
                        "EndpointIP": "10.0.5.24",
                        "Info": {
                            "Host IP": "172.24.32.34"
                        }
                    }
                ]
            }
        }
    }
]
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:       17.12.1-ce
 API version:   1.35
 Go version:    go1.9.4
 Git commit:    7390fc6
 Built: Tue Feb 27 22:17:40 2018
 OS/Arch:       linux/amd64

Server:
 Engine:
  Version:      17.12.1-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   7390fc6
  Built:        Tue Feb 27 22:16:13 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Containers: 3
 Running: 3
 Paused: 0
 Stopped: 0
Images: 3
Server Version: 17.12.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: tq0iq3bj26qeirhfmsw6ud8x4
 Is Manager: true
 ClusterID: ivj9b4ikawj9wq4g3gztl3z4l  
 Managers: 5
 Nodes: 9
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0 
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 172.24.35.240
 Manager Addresses:
  172.24.32.156:2377
  172.24.32.226:2377
  172.24.35.240:2377
  172.24.35.83:2377
  172.24.36.37:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9b55aab90508bd389d7654c4baf173a981477d55
runc version: 9f9c96235cc97674e935002fc3d78361b696a69e
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-1049-aws
Operating System: Ubuntu 16.04.3 LTS   
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858GiB
Name: omni-sandbox-manager-g1<DOMAIN_REDACTED>
ID: 6UHL:CALE:AG7E:C7UK:EDUQ:CYQZ:3TM7:GIFE:IYAA:LVG4:2FU4:LRN3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/  
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

The environment is AWS, built via terraform + ansible.  The swarm details above are from a freshly instantiated swarm that had only been running for a few minutes and had just been deployed to once.

**Comment by jaschaio** [comment]
I got the same issue as in #24186 

I actually noticed that while looking into an issue with [docker flow proxy](https://github.com/vfarcic/docker-flow-proxy/issues/454)

I set up the hosts today.

**Docker Info:**
```
Containers: 6
 Running: 2
 Paused: 0
 Stopped: 4
Images: 7
Server Version: 18.03.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: vqbiycm98ydyufdjpwi0eacu2
 Is Manager: true
 ClusterID: nfa5qg4moenuuzuiplcxrn2dm
 Managers: 1
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 10.136.65.80
 Manager Addresses:
  10.136.65.80:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfd04396dc68220d1cecbe686a6cc3aa5ce3667c
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-116-generic
Operating System: Ubuntu 16.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 992.2MiB
Name: Docker-Manager
ID: NY56:PHXZ:OBLW:CA2B:CBIF:ESP5:MSYI:ACT2:MX67:NVMA:FJMF:PSBJ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Docker Version:**
```
Client:
 Version:	18.03.0-ce
 API version:	1.37
 Go version:	go1.9.4
 Git commit:	0520e24
 Built:	Wed Mar 21 23:10:01 2018
 OS/Arch:	linux/amd64
 Experimental:	false
 Orchestrator:	swarm

Server:
 Engine:
  Version:	18.03.0-ce
  API version:	1.37 (minimum version 1.12)
  Go version:	go1.9.4
  Git commit:	0520e24
  Built:	Wed Mar 21 23:08:31 2018
  OS/Arch:	linux/amd64
  Experimental:	false
```

Output of `docker network inspect proxy -v` on `manager-node`
```
[
    {
        "Name": "proxy",
        "Id": "euybel6l84tolld9u6elkzyha",
        "Created": "2018-04-04T08:35:34.100622456-04:00",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.0.0/24",
                    "Gateway": "10.0.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "5f0529846e6f160c681f444f3e545ba91de35ba9a60884625b4367a0af2e0275": {
                "Name": "proxy.1.torntlflfob5txv30fpu7c2as",
                "EndpointID": "561ccc3e25e9c8a0fc1fb5d6f38bd44e703357998541715ce4ebf7eba795d52c",
                "MacAddress": "02:42:0a:00:00:0c",
                "IPv4Address": "10.0.0.12/24",
                "IPv6Address": ""
            },
            "97f81278baca67d546577cc77f2ef2787af3ff6dc1dec584955a849420c90b1d": {
                "Name": "swarm-listener.1.lpx454l7w5sad90hhbxdb0g1m",
                "EndpointID": "74759d561dbca129e70f6d530a1a185a4bd3aa7280e4561f7e123bf06c300189",
                "MacAddress": "02:42:0a:00:00:06",
                "IPv4Address": "10.0.0.6/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4103"
        },
        "Labels": {},
        "Peers": [
            {
                "Name": "ec06bceccb17",
                "IP": "10.136.65.80"
            }
        ],
        "Services": {
            "proxy": {
                "VIP": "10.0.0.11",
                "Ports": [],
                "LocalLBIndex": 376,
                "Tasks": [
                    {
                        "Name": "proxy.1.torntlflfob5txv30fpu7c2as",
                        "EndpointID": "561ccc3e25e9c8a0fc1fb5d6f38bd44e703357998541715ce4ebf7eba795d52c",
                        "EndpointIP": "10.0.0.12",
                        "Info": {
                            "Host IP": "10.136.65.80"
                        }
                    }
                ]
            },
            "swarm-listener": {
                "VIP": "10.0.0.5",
                "Ports": [],
                "LocalLBIndex": 374,
                "Tasks": [
                    {
                        "Name": "swarm-listener.1.lpx454l7w5sad90hhbxdb0g1m",
                        "EndpointID": "74759d561dbca129e70f6d530a1a185a4bd3aa7280e4561f7e123bf06c300189",
                        "EndpointIP": "10.0.0.6",
                        "Info": {
                            "Host IP": "10.136.65.80"
                        }
                    }
                ]
            }
        }
    }
]
```

Output of `docker network inspect proxy -v` on a `worker-node`

```
[
    {
        "Name": "proxy",
        "Id": "euybel6l84tolld9u6elkzyha",
        "Created": "2018-04-04T14:05:19.185670628-04:00",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.0.0/24",
                    "Gateway": "10.0.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "30b50db2fd98cb1adbbff8ba47e67296c488b2310f6193d283bfcd6fb7255db4": {
                "Name": "nginx-test.1.cbv71zm337yb4y0abve2p54jt",
                "EndpointID": "ae2ace61688e044262c008bebe36d19ce21b1594065db3775834a1cfe54b99da",
                "MacAddress": "02:42:0a:00:00:1e",
                "IPv4Address": "10.0.0.30/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4103"
        },
        "Labels": {},
        "Peers": [
            {
                "Name": "c136763babb4",
                "IP": "10.136.81.222"
            }
        ],
        "Services": {
            "nginx-test": {
                "VIP": "10.0.0.29",
                "Ports": [],
                "LocalLBIndex": 288,
                "Tasks": [
                    {
                        "Name": "nginx-test.1.cbv71zm337yb4y0abve2p54jt",
                        "EndpointID": "ae2ace61688e044262c008bebe36d19ce21b1594065db3775834a1cfe54b99da",
                        "EndpointIP": "10.0.0.30",
                        "Info": {
                            "Host IP": "10.136.81.222"
                        }
                    }
                ]
            }
        }
    }
]
```

**Comment by abhi** [comment]
Fixed by https://github.com/moby/moby/pull/37045

**Comment by pa-yourserveradmin-com** [comment]
It looks like I'm able to see the same issue right now, while this issue is closed and changes from https://github.com/moby/moby/pull/37045 were released in [18.06.0-ce](https://docs.docker.com/engine/release-notes/18.06/#18060-ce).

Symptoms are the same: `docker network inspect -v` returns the same output information as without `-v` flag and this information lacks details about containers and services running on Swarm workers when there is no task related to the same network scheduled on managers.

Tested with `19.03.8` and `18.06.2-ce` (client and server versions are the same).

Please let me know if new issue should be created here.

**Comment by IgorOhrimenko** [comment]
I have the same issue on the 19.03.13.
`docker network inspect` contained no container:
`        "Containers": null,`
and no peers section.

I fixed it like this:
docker stack rm stackname
docker stack deploy --compose-file docker-compose.yaml stackname
